modules/axipi/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+//! [ArceOS](https://github.com/arceos-org/arceos) Inter-Processor Interrupt (IPI) primitives.
+
+#![cfg_attr(not(test), no_std)]
+
+#[macro_use]
+extern crate log;
+extern crate alloc;
+
+use lazyinit::LazyInit;
+
+use kspin::SpinNoIrq;
+
+use axhal::irq::{IpiTarget, IPI_IRQ};
+use axhal::percpu::this_cpu_id;
+
+mod event;
+mod queue;
+
+pub use event::*;
+use queue::IPIEventQueue;
+
+#[percpu::def_percpu]
+static IPI_EVENT_QUEUE: LazyInit<SpinNoIrq<IPIEventQueue>> = LazyInit::new();
+
+/// Initialize the per-CPU IPI event queue.
+pub fn init() {
+    IPI_EVENT_QUEUE.with_current(|ipi_queue| {
+        ipi_queue.init_once(SpinNoIrq::new(IPIEventQueue::default()));
+    });
+}
+
+/// Sends an IPI event to the processor(s) specified by `dest_cpu`.
+pub fn send_ipi_one<T: Into<Callback>>(dest_cpu: usize, callback: T) {
```

Comment:
This function essentially runs the `callback` on `dest_cpu`, rename it to `run_on_cpu`?